### PR TITLE
test: add integration tests for audit service

### PR DESCRIPTION
**Pull request by htztomic**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/example/app/src/routes/auditRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      } catch (err) {
+        if (isAuditIncompleteError(err)) {
+          throw Boom.badRequest(err.message);
+        }
+        throw err;
+      }
+    })
+  );
+
+  router.post(
```
Comment (DaveMadsen) [inline workbench-core/example/app/src/routes/auditRoutes.ts:47]: I probably would have made this a `router.get()`

File: workbench-core/example/app/src/routes/auditRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import { Request, Response, Router } from 'express';
+import { wrapAuth } from '../utilities/authWrapper';
+
+export function setupAuditRoutes(
+  router: Router,
+  auditService: AuditService,
+  dynamicAuthorizationService: DynamicAuthorizationService
+): void {
+  router.post(
+    '/audit/write',
```
Comment (DaveMadsen) [inline workbench-core/example/app/src/routes/auditRoutes.ts:20]: probably don't need `write` in this path

File: workbench-core/example/infrastructure/package.json [hunk 3]
```diff
@@ -98,8 +98,10 @@
     "eslint": "^8.7.0",
     "eslint-plugin-security": "^1.4.0",
     "express": "^4.17.3",
+    "fast-check": "^2.24.0",
     "istanbul-badges-readme": "1.8.1",
     "jest": "^27.5.1",
+    "jest-fast-check": "^1.0.2",
```
Comment (DaveMadsen) [inline workbench-core/example/infrastructure/package.json:104]: are you actually using these new imports?

File: workbench-core/example/app/src/configs/dynamicRouteConfig.ts [hunk 4]
```diff
@@ -62,6 +62,28 @@ export const dynamicRoutesMap: DynamicRoutesMap = {
...
+          subjectId: '*',
+          subjectType: 'auditEntry'
+        }
+      }
+    ]
+  },
+  '/audit/is-audit-complete': {
+    POST: [
+      {
+        action: 'CREATE',
```
Comment (DaveMadsen) [inline workbench-core/example/app/src/configs/dynamicRouteConfig.ts:80]: I would probably have made this route a GET with a READ permission

**Review by DaveMadsen — COMMENTED** [review]
I added a few comments. Mostly optional items that correspond to my personal preference and non-blocking.
But please double check `package.json` and determine if those imports of `fast-check` and `jest-fast-check` are indeed needed. I may have missed it, but I didn't see any tests using them.

File: workbench-core/example/app/src/configs/dynamicRouteConfig.ts (hunk 4 above)
Comment (htztomic) [reply workbench-core/example/app/src/configs/dynamicRouteConfig.ts:80]: changed

File: workbench-core/example/infrastructure/package.json (hunk 3 above)
Comment (htztomic) [reply workbench-core/example/infrastructure/package.json:104]: using to randomize objects

File: workbench-core/example/app/src/routes/auditRoutes.ts (hunk 2 above)
Comment (htztomic) [reply workbench-core/example/app/src/routes/auditRoutes.ts:20]: fixed

File: workbench-core/example/app/src/routes/auditRoutes.ts (hunk 1 above)
Comment (htztomic) [reply workbench-core/example/app/src/routes/auditRoutes.ts:47]: fixed

**Review by DaveMadsen — APPROVED** [review]

**Pull request merged**